hutch_python/env_version.py

```diff
@@ -51,11 +57,12 @@ def get_standard_dev_pkgs() -> set[str]:
     pythonpath = os.environ.get('PYTHONPATH', '')
     if not pythonpath:
         return set()
-    pkg_names = set()
-    for part in pythonpath.split(':'):
-        if any(((s in part) for s in _dev_ignore_list)) or (not part):
-            continue
-        pkg_names.add(os.path.basename(os.path.dirname(part)))
+    paths = pythonpath.split(':')
+    valid_paths = filter(not_ignored, paths)
+    pkg_names = set(n.name for n in
+                    pkgutil.iter_modules(path=valid_paths)
```

Comment:
I think this was the best possible approach, leveraging the normal python package tools